httputil/client.go

```diff
@@ -33,6 +41,58 @@ type ClientOptions struct {
 	Proxy      func(*http.Request) (*url.URL, error)
 }
 
+func addLocalSslCertificates(conf *tls.Config) (allCAs *x509.CertPool, err error) {
+	if conf != nil && conf.RootCAs != nil {
+		allCAs = conf.RootCAs
+	} else {
+		allCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("cannot read system certificates: %v", err)
+		}
+	}
+	if allCAs == nil {
+		return nil, fmt.Errorf("cannot use empty certificate pool")
+	}
+	extraCertFiles, err := filepath.Glob(filepath.Join(dirs.SnapdExtraSslCertsDir, "*.pem"))
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range extraCertFiles {
+		extraCert, err := ioutil.ReadFile(p)
```

Comment:
I wonder if this reading again and again will be a problem, or just marginal

Reply:
I added a XXX

Reply:
It seems to strike the right balance to cache these in memory and require a daemon restart to pick up changes to added store certs.

Reply:
if we feel we need a cache (which we might and it might be  a case where it's easier to just do it than measure),  then httputil is the wrong place to implement it though: so the API from httputil should just be to take something like:

```
type ExtraSSLCerts interface {
    Certs() ...
}
```
Some other part of snapd will implement this plus suitable Add* and Remove* to use from configcore.

To be clear we don't expect the directory itself to be an official API where people drop file, it will mostly exist because storing certs in state.json will get unwieldy.

Reply:
I added an abstraction now